okio/src/main/java/okio/Buffer.java

```diff
@@ -164,30 +164,26 @@ public Buffer copyTo(Buffer out, long offset, long byteCount) {
     checkOffsetAndCount(size, offset, byteCount);
     if (byteCount == 0) return this;
 
-    Segment source = head;
-    Segment target = out.writableSegment(1);
     out.size += byteCount;
 
-    while (byteCount > 0) {
-      // If necessary, advance to a readable source segment. This won't repeat after the first copy.
-      while (offset >= source.limit - source.pos) {
-        offset -= (source.limit - source.pos);
-        source = source.next;
-      }
+    // Skip segments that we aren't copying from.
+    Segment s = head;
+    for (; offset >= (s.limit - s.pos); s = s.next) {
+      offset -= (s.limit - s.pos);
+    }
 
-      // If necessary, append another target segment.
-      if (target.limit == Segment.SIZE) {
-        target = target.push(SegmentPool.INSTANCE.take());
+    // Copy one segment at a time.
+    for (; byteCount > 0; s = s.next) {
+      Segment copy = new Segment(s);
```

Comment:
I think this is the most subtle part of the code. It isn't really the concern of this method whether it is a copy of the input data, or a view of it. I think those new to okio post-merge might not assume copy meant copy of data. For these reasons, and the fact that I can't think of a better word than copy, LG